module/zfs/zil.c

```diff
@@ -1269,21 +1290,21 @@ zil_commit_waiter_link_lwb(zil_commit_waiter_t *zcw, lwb_t *lwb)
 {
 	/*
 	 * The lwb_waiters field of the lwb is protected by the zilog's
-	 * zl_lock, thus it must be held when calling this function.
+	 * zl_issuer_lock while the lwb is open and zl_lock otherwise.
+	 * zl_issuer_lock also protects leaving the open state.
+	 * zcw_lwb setting is protected by zl_issuer_lock and state !=
+	 * flush_done, which is protected by zl_lock.
```

Comment:
This sentence is confusing to me; setting zcw_lwb is protected by the zl_issuer lock, but I'm not sure how the `and state != flush_done...` clause relates. Is it saying it's protected by the lock if state != flush done, and the state change is protected by the zl_lock?

Reply:
zil_commit_waiter_link_lwb() can never be called if state == flush_done, which transition is protected by zl_lock.  In other cases zcw_lwb is protected by combination of zl_issuer_lock and zcw_lock, but zil_commit_waiter_link_lwb() cares only about the first, since the second is used only for zcw_lwb clearing.

Reply:
Added word "transition" to the comment.